addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js

```diff
@@ -12,46 +12,58 @@ export class ConfirmationDialog extends Component {
         this.modalRef = useChildRef();
         this.isConfirmedOrCancelled = false; // ensures we do not confirm and/or cancel twice
     }
+
+    get context() {
+        return {};
+    }
+
     async _cancel() {
-        if (this.isConfirmedOrCancelled) {
-            return;
+        this.execButton(this.props.cancel);
+    }
+
+    async _confirm() {
+        this.execButton(this.props.confirm);
+    }
+
+    disableButtons() {
+        if (!this.modalRef.el) {
+            return; // safety belt for stable versions
         }
-        this.isConfirmedOrCancelled = true;
-        this.disableButtons();
-        if (this.props.cancel) {
-            try {
-                await this.props.cancel();
-            } catch (e) {
-                this.props.close();
-                throw e;
-            }
+        for (const button of [...this.modalRef.el.querySelectorAll(".modal-footer button")]) {
+            button.disabled = true;
         }
-        this.props.close();
     }
-    async _confirm() {
+
+    enableButtons() {
+        if (!this.modalRef.el) {
+            return; // safety belt for stable versions
+        }
+        for (const button of [...this.modalRef.el.querySelectorAll(".modal-footer button")]) {
+            button.disabled = false;
+        }
+    }
```

Comment:
fuse these and do setButtonsDisabled(true/flase). You can also remove the safety belt for stable